buffer.go

```diff
@@ -109,39 +109,39 @@ func (b *buffer) readNext(need int) ([]byte, error) {
 // If possible, a slice from the existing buffer is returned.
 // Otherwise a bigger buffer is made.
 // Only one buffer (total) can be used at a time.
-func (b *buffer) takeBuffer(length int) []byte {
+func (b *buffer) takeBuffer(length int) ([]byte, error) {
 	if b.length > 0 {
-		return nil
+		return nil, ErrUnreadTxRows
```

Comment:
You're right. Actually we can get this error only when we're trying to write some packet. I'm not sure how to determine that we're in a tx the only way I see to override `writeCommandPacket*` functions for `mysqlTx`